internal/helm/repository.go

```diff
@@ -61,69 +61,85 @@ func NewChartRepository(repositoryURL string, providers getter.Providers, opts [
 }
 
 // Get returns the repo.ChartVersion for the given name, the version is expected
-// to be a semver.Constraints compatible string. If version is empty, the latest
-// stable version will be returned and prerelease versions will be ignored.
+// to be a semver.Constraints compatible string. In case of an ambiguous match,
+// takes the freshest chart. If version is empty, the latest stable version will
+// be returned and prerelease versions will be ignored.
 func (r *ChartRepository) Get(name, version string) (*repo.ChartVersion, error) {
-	cvs, ok := r.Index.Entries[name]
+	chartVersions, ok := r.Index.Entries[name]
 	if !ok {
 		return nil, repo.ErrNoChartName
 	}
-	if len(cvs) == 0 {
+	if len(chartVersions) == 0 {
 		return nil, repo.ErrNoChartVersion
 	}
 
-	// Check for exact matches first
-	if len(version) != 0 {
-		for _, cv := range cvs {
-			if version == cv.Version {
-				return cv, nil
-			}
-		}
-	}
-
-	// Continue to look for a (semantic) version match
-	latestStable := len(version) == 0 || version == "*"
-	var match semver.Range
-	if !latestStable {
-		rng, err := semver.ParseRange(version)
+	// Parse version constraints if given (default strategy: latest)
+	var versionLatestStable = len(version) == 0 || version == "*"
```

Comment:
Better to fall back to `semver.NewConstraint("*")`, to piggy back on the library's logic.